cid/export.py

```diff
@@ -96,109 +139,233 @@ def export_analysis(qs):
+            locations = re.findall("LOCATION\n  '(s3://.+?)'", view_data.get('data'))
+            for location in locations:
+                logger.info(f'Please replace manually location bucket with a parameter: s3://{location}')
+                default = get_parameter(
+                    f'{key}-s3path',
+                    'please provide default value',
+                    default=location,
+                )
+                view_data['parameters'] = {
+                    f'{key}-s3path': {
+                        'default': default,
+                        'description': f"S3 Path for {key} table",
+                    }
+                }
+                view_data['data'] = view_data['data'].replace(location, '${s3path}')
 
-    reader_account_id = get_parameter(
-        'reader-account',
-        message='Enter account id to share the template with or *',
-        default='*'
-    )
-    res = qs.update_template_permissions(
-        TemplateId=template_id,
-        GrantPermissions=[
-            {
-                "Principal": f'arn:aws:iam::{reader_account_id}:root' if reader_account_id != '*' else '*',
-                'Actions': [
-                    "quicksight:DescribeTemplate",
-                ]
-            },
-        ],
-    )
+        resources['views'][key] = view_data
 
+    logger.debug('Building dashboard resource')
     dashboard_id = get_parameter(
         'dashboard-id',
         message='dashboard id (will be used in url of dashboard)',
-        default=analysis['Name'].replace(' ', '-').lower()
+        default=escape_id(analysis['Name'].lower())
     )
 
-    resources = {}
-    resources['dashboards'] = {}
-    resources['datasets'] = {}
-    resources['views'] = {}
+    dashboard_resource = {}
+    dashboard_resource['dependsOn'] = {
+        # Historically CID uses dataset names as dataset reference. IDs of manually created resources have uuid format.
+        # We can potentially reconsider this and use IDs at some point
+        'datasets': sorted(list(set([dataset_name for dataset_name in datasets.keys()] + resources_datasets))) 
+    }
+    dashboard_resource['name'] = analysis['Name']
+    dashboard_resource['dashboardId'] = dashboard_id
 
-    resources['dashboards'][analysis['Name'].upper()] = {
-        'name': analysis['Name'],
-        'templateId': template_id,
-        'sourceAccountId': qs.account_id,
-        'region': qs.session.region_name,
-        'dashboardId': dashboard_id,
-        'dependsOn':{
-            'datasets': [dataset['Name'].replace(' ', '-') for dataset in datasets.values()]
+    dashboard_export_method = None
+    if get_parameters().get('template-id'):
+        dashboard_export_method = 'template'
+    else:
+        dashboard_export_method = get_parameter(
+            'dashboard-export-method',
+            message='Please choose export method (pull a json "definition" or create a "template")',
+            choices=[
+                'definition',
```

Comment:
Suggest to change to:
template definition file
shared template in QuickSight account

Reply:
not sure to understand what is requested. can you provide full line with your option? will be happy to apply.

Reply:
These must be short as the same values will be used in command line parameters

Reply:
I can probably rewrite this to achieve the same results

Reply:
done
<img width="660" alt="image" src="https://user-images.githubusercontent.com/82834333/228655389-b6358497-94d5-4823-9127-7b0cd28834c4.png">